base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/configuration/ConfigInfo.java

```diff
@@ -148,6 +148,36 @@ public String apply(ConfigInfo info) {
         
         return namesWithProtectionFlag;
     }
+    
+    /**
+     * Returns Configuration/Component names and their protection status, excluding the current configuration
+     * @param infos
+     * @return collection of pair config/comp Name and Protection Flag (e.g. {<name>=<prot. status>, ...}), excluding the current configuration
+     */
+    public static Map<String, Boolean> mapNamesWithTheirProtectionFlagWithoutCurrent(Collection<ConfigInfo> infos) {
+    	Map<String, Boolean> filteredResults = mapNamesWithTheirProtectionFlag(infos);
+    	filteredResults.remove(Configurations.getInstance().display().displayCurrentConfig().getValue().name());
+    	return filteredResults;
+    }
+    
+    /**
+     * Checks if there is a protected configuration/comp within the given ConfigInfos objects
+     * @param infos
+     * @return boolean: True if a protected configuration exists within given infos, False if not
+     */
+    public static Boolean hasProtectedConfigs(Collection<ConfigInfo> infos) {
+        if (infos == null) {
```

Comment:
Infos will never be null as it is always assigned in the constructor. However, It can be empty, so we should better check if it is empty.